src/Thruway/Role/Subscriber.php

```diff
@@ -128,6 +134,32 @@ protected function processSubscribed(ClientSession $session, SubscribedMessage $
             }
         }
     }
+    
+    /**
+     * Process unsubscribe error
+     * 
+     * @param \Thruway\AbstractSession $session
+     * @param \Thruway\Message\ErrorMessage $msg
+     */
+    protected function processUnsubscribeError(AbstractSession $session, ErrorMessage $msg)
+    {
+        foreach ($this->subscriptions as $key => $subscription) {
+            if ($subscription["unsubscribed_request_id"] === $msg->getErrorRequestId()) {
+                // reject the promise
+                $subscription['unsubscribed_deferred']->reject($msg);
```

Comment:
This deferred is the one that wraps the promise returned by the `unsubscribe` method, so yes, it is used (and is kind of important.) Or I didn't understand what you meant?